### ISSUE TITLE
WebSocketBadStatusException: Handshake status 200 OK

### ISSUE THREAD

**Issue by dawidwekwejt** [mandrewcito/signalrcore#46]
I got Handshake status 200 OK instead of one success statuses.

**LOGS**
```
2020-11-16 21:37:58,797 - SignalRCoreClient - DEBUG - Handler registered started AwakeReceived_Group
2020-11-16 21:37:58,797 - SignalRCoreClient - DEBUG - Negotiate url:https://............/hubs/awake/negotiate
c:\python\lib\site-packages\urllib3\connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host '............'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
2020-11-16 21:37:58,984 - SignalRCoreClient - DEBUG - Response status code200
2020-11-16 21:37:58,985 - SignalRCoreClient - DEBUG - Connection started
2020-11-16 21:37:58,985 - SignalRCoreClient - DEBUG - start url:wss://............/hubs/awake?id=NQZMZmxqk0q5EjPLkdNORA
2020-11-16 21:37:59,063 - websocket - DEBUG - --- request header ---
2020-11-16 21:37:59,063 - websocket - DEBUG - GET /hubs/awake?id=NQZMZmxqk0q5EjPLkdNORA HTTP/1.1
Upgrade: websocket
Host: ............
Origin: http://............
Sec-WebSocket-Key: dedr/jpAeI8mTzdqfO9gNw==
Sec-WebSocket-Version: 13
Connection: upgrade


2020-11-16 21:37:59,064 - websocket - DEBUG - -----------------------
2020-11-16 21:37:59,064 - websocket - DEBUG - --- response header ---
2020-11-16 21:37:59,084 - websocket - DEBUG - HTTP/1.1 200 OK
2020-11-16 21:37:59,084 - websocket - DEBUG - Content-Length: 0
2020-11-16 21:37:59,084 - websocket - DEBUG - Content-Type: application/octet-stream
2020-11-16 21:37:59,085 - websocket - DEBUG - Server: Microsoft-IIS/8.5
2020-11-16 21:37:59,085 - websocket - DEBUG - X-Powered-By: ASP.NET
2020-11-16 21:37:59,085 - websocket - DEBUG - Date: Mon, 16 Nov 2020 20:37:58 GMT
2020-11-16 21:37:59,085 - websocket - DEBUG - -----------------------
2020-11-16 21:37:59,086 - SignalRCoreClient - DEBUG - -- web socket error --
2020-11-16 21:37:59,092 - SignalRCoreClient - ERROR - Traceback (most recent call last):
  File "c:\python\lib\site-packages\websocket\_app.py", line 249, in run_forever
    self.sock.connect(
  File "c:\python\lib\site-packages\websocket\_core.py", line 226, in connect
    self.handshake_response = handshake(self.sock, *addrs, **options)
  File "c:\python\lib\site-packages\websocket\_handshake.py", line 80, in handshake
    status, resp = _get_resp_headers(sock)
  File "c:\python\lib\site-packages\websocket\_handshake.py", line 165, in _get_resp_headers
    raise WebSocketBadStatusException("Handshake status %d %s", status, status_message, resp_headers)
websocket._exceptions.WebSocketBadStatusException: Handshake status 200 OK

2020-11-16 21:37:59,093 - SignalRCoreClient - ERROR - <signalrcore.hub.base_hub_connection.BaseHubConnection object at 0x0000027025E8B850> Handshake status 200 OK
2020-11-16 21:37:59,093 - SignalRCoreClient - ERROR - Handshake status 200 OK <class 'websocket._exceptions.WebSocketBadStatusException'>
connection closed
2020-11-16 21:37:59,093 - websocket - ERROR - error from callback <bound method BaseHubConnection.on_socket_error of <signalrcore.hub.base_hub_connection.BaseHubConnection object at 0x0000027025E8B850>>: Handshake status 200 OK
  File "c:\python\lib\site-packages\websocket\_app.py", line 344, in _callback
    callback(*args)
  File "c:\python\lib\site-packages\signalrcore\hub\base_hub_connection.py", line 162, in on_socket_error
    raise HubError(error)
2020-11-16 21:37:59,097 - SignalRCoreClient - DEBUG - -- web socket close --
connection closed
depth mask updated
```

**Comment by mandrewcito** [comment]
Hi,

This issue could be related with that? 
https://stackoverflow.com/questions/45840855/websocket-handshake-status-200-exception

**Comment by dawidwekwejt** [comment]
> Hi,
> 
> This issue could be related with that?
> https://stackoverflow.com/questions/45840855/websocket-handshake-status-200-exception

Hi
My site is over HTTPS so WSS protocol for web sockets connection is right. Do your library use only websockets or is there possiblility to connect over long pooling?

**Comment by mandrewcito** [comment]
only websockets for now.

**Comment by dawidwekwejt** [comment]
> only websockets for now.
I think I found my misstake. My problem has palce on my company network and application infrastucture. SignalR have all transport types turn on by default. I this case I got "Handshake status 200 OK". When I set transport type only to WebSockets I got "Handshake status 404 Not Found". My thesis is our IIS is not configured to allow websocket communication.
I'll write to our administratiors to set this feature in IIS. I'll write when I have some new informations.
